component/exporter.go

```diff
@@ -157,6 +157,13 @@ func WithLogsExporter(createLogsExporter CreateLogsExporterFunc) ExporterFactory
 	}
 }
 
+// WithExporterStabiityLevel overrides the default "unmaintained" stability level.
+func WithExporterStabilityLevel(sl map[config.Type]StabilityLevel) ExporterFactoryOption {
+	return func(o *exporterFactory) {
+		o.stability = sl
+	}
+}
+
 // NewExporterFactory returns a ExporterFactory.
 func NewExporterFactory(cfgType config.Type, createDefaultConfig ExporterCreateDefaultConfigFunc, options ...ExporterFactoryOption) ExporterFactory {
```

Comment:
Using a map makes sense to me, but someone can still do

```go
component.NewExporterFactory(
	typeStr,
	createDefaultConfig,
	component.WithTracesExporter(createTracesExporter),
	component.WithExporterStabilityLevel(map[config.Type]component.StabilityLevel{
		config.TracesDataType:  component.StabilityLevelStable,
		config.MetricsDataType: component.StabilityLevelStable, // WRONG: no support for metrics
}))
```

I think we should detect and error out on this case